src/main/java/com/avispl/symphony/dal/communicator/barco/e2gen2/BarcoE2Communicator.java

```diff
@@ -572,6 +583,12 @@ private String getActivePresetName() throws Exception {
 		if (presetResponse == null) {
 			return BarcoE2Constant.NONE;
 		}
+		if (presetResponse.get(BarcoE2Constant.SUCCESS) != null && presetResponse.get(BarcoE2Constant.SUCCESS).asInt() != 0) {
```

Comment:
Both presetResponse.get(BarcoE2Constant.SUCCESS) and response.get(BarcoE2Constant.RESPONSE) may be extracted into a variable, so each of them is only extracted once

Reply:
Fixed